torch/_dynamo/skipfiles.py

```diff
@@ -208,51 +255,46 @@ def add(import_name: str):
     _recompile_re()
 
 
+@dataclasses.dataclass
+class SkipResult:
+    skipped: bool
+    reason: Optional[str]
+
+
 def check(filename, allow_torch=False):
     """Should skip this file?"""
     if filename is None:
-        return True
+        return SkipResult(True, "filename is None")
     if filename in FILENAME_ALLOWLIST:
-        return False
-    if allow_torch and is_torch(filename):
-        return False
+        return SkipResult(
+            False,
+            "allowlisted in skipfiles.FILENAME_ALLOWLIST",
+        )
+    if allow_torch and is_torch_inline_allowed(filename):
+        return SkipResult(
+            False,
+            "allowlisted in skipfiles.SUBMODULE_ALLOWLIST",
+        )
     if is_fbcode and bool(FBCODE_SKIP_DIRS_RE.match(filename)):
-        return True
-    return bool(SKIP_DIRS_RE.match(filename))
+        return SkipResult(
+            True,
+            "should be skipped according skipfiles.FBCODE_SKIP_DIRS",
+        )
+    if bool(SKIP_DIRS_RE.match(filename)):
+        return SkipResult(True, "should be skipped according skipfiles.SKIP_DIRS")
+    else:
+        return SkipResult(False, "inlining by default")
```

Comment:
For the ```check``` function, we can eventually remove the ```allow_torch``` argument in the future. There are two places to call this function, 1/ in ```eval_frame.py``` where we start tracing a frame (allow_torch=False); 2/ in ```symbolic_convert.py``` where we plan to inline a function (allow_torch=True). I didn't see too much difference between these two cases. The prototype I did hit several latent bugs in our codebase, so I'll leave it to a separate PR - 3/N.